source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -42,14 +42,17 @@ constexpr auto filter_metadata_key = "com.amazonaws.lambda";
 constexpr auto egress_gateway_metadata_key = "egress_gateway";
 
 void setLambdaHeaders(Http::RequestHeaderMap& headers, const absl::optional<Arn>& arn,
-                      InvocationMode mode) {
+                      InvocationMode mode, const std::string& host_rewrite) {
   headers.setMethod(Http::Headers::get().MethodValues.Post);
   headers.setPath(fmt::format("/2015-03-31/functions/{}/invocations", arn->arn()));
   if (mode == InvocationMode::Synchronous) {
     headers.setReference(LambdaFilterNames::get().InvocationTypeHeader, "RequestResponse");
   } else {
     headers.setReference(LambdaFilterNames::get().InvocationTypeHeader, "Event");
   }
+  if (!host_rewrite.empty()) {
+    headers.setHost(host_rewrite);
```

Comment:
I'm wondering if this makes sense to do here for the Envoy headers (vs. the signed headers). Basically, if we setHost here and don't clear route cache it works fine. But if we have a filter following AWS lambda, and it does clear route cache for some reason, we would use this host value to lookup in RouteConfiguration. 

Maybe we should just ask people to set the hostRewrite in two places, here and HCM? That doesn't feel great though, since it's not single source of truth.

Another option could be making this an upstream filter - logically this is not intended to influence routing right?

Reply:
I agree, this filter make more sense to be an upstream filter. it depends on the cluster. 
I don't get you point on the routing. but I think it is a case of having a filter after this one. for example, adding a lua filter that change any header involve in the signature will break the signature. and adding weight routing with 2 clusters, one on us-east and the other on us-west, will fall into a situation that depending on the cluster|arn sometime the request will fails and sometimes it will not.

not making this filter an upstream filter seems to have many constraints at the moment to configure it.

saying that, in the use case we have, the big problem is we use the auto_host_rewrite in the route configuration. adding the upstream host in the configuration like aws_request_signature is the lower impact change that could help in our scenario.

regarding
> I'm wondering if this makes sense to do here for the Envoy headers (vs. the signed headers).
I'm new in envoy and don't know some of the conceps in the core and routing implementation. can you explains what you mean? is there other set of headers?

Reply:
Adding Info, I did the check converting the filter in an upstream filter, but at the moment it is invoked, the host header is still not replaced, I see it happens after those filters are called, then the final value cannot be used. Is this expected?

Reply:
I guess the question is whether the route cache should be cleared, to force consistency of the selected header with the chosen route? If we don't go this path, I think the only sensible guidance will be to tell users to place this filter after any other filter that could possible clear route cache. @yanavlasov any thoughts on other filters that look like this, where host is rewritten?

Reply:
there is another issue with this filter, that is related to change headers, :path https://github.com/envoyproxy/envoy/issues/32101

I just check other option to rewrite it (added a comment in the issue) and when I check in the code, there is a comment indicating it happens when headers are changed without clearing the route cache

https://github.com/envoyproxy/envoy/blob/main/source/common/router/config_impl.cc#L1560

apology for my ignorance, what is exactly the "clearing route cache"

Reply:
probably instead of this change make more sense to convert the filter to upstream filter and see if there is a bug that auto_host_rewrite is not replacing the host header before calling upstream filters. host_rewrite_literal is doing it before calling it.

Reply:
Clearing the route cache applies for downstream filters. HCM will evaluate the `RouteConfiguration` lookup and "cache" the result" at the start of filter chain evaluation. Any filter that modifies fields that affect route lookup can trigger a fresh lookup via https://github.com/envoyproxy/envoy/blob/0edcdf8639455605392b9191d4a9b5576703d951/envoy/http/filter.h#L310.

Reply:
In any case, it looks like you're aiming for consistentcy with the request signing filter which behaves this way, so we can split out the issues I'm flagging from this review. 

Will defer to @suniltheta for LGTM on first pass.

Reply:
The action of re-evaluating the route (poorly names as "clear route cache) is normally configured as part of the header mutation (i.e. in ext_proc or ext_authz).

I looks like in this case the behavior can be documented, such that re-evaluating the route after changing host can result in unexpected behavior.

Reply:
@yanavlasov Do I need to add something to the current documentation?

Reply:
I would just add a note stating that changing the value of the host header can result in a different route to be selected if an HTTP filter after AWS lambda re-evaluates the route (clears route cache).

Reply:
Done. Have added the note in the API to be available in the doc.